internal/api/client.go

```diff
@@ -17,25 +20,50 @@ type ClientConfig struct {
 	AccessToken      string
 	OrganizationSlug string
 	ServerBaseUrl    string
+	DebugEnabled     bool
 }
 
 // authTransport is a middleware for the HTTP client.
 type authTransport struct {
-	accessToken string
+	accessToken  string
+	debugEnabled bool
 }
 
 // RoundTrip adds the Authorization header to all requests made by the HTTP client.
 func (t *authTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	if t.debugEnabled {
+		fmt.Println("--- 🕵️ DEBUG")
+		fmt.Println("Request:")
+		fmt.Printf("%s %s\n\n", req.Method, req.URL.String())
+	}
+
+	// IMPORTANT: We need to set the token after printing the request to avoid leaking the token in logs.
 	req.Header.Set("Authorization", "Bearer "+t.accessToken)
-	return http.DefaultTransport.RoundTrip(req)
+
+	resp, err := http.DefaultTransport.RoundTrip(req)
+
+	if t.debugEnabled {
+		body, _ := io.ReadAll(resp.Body)
+		defer resp.Body.Close()
+
+		// We need to replace the body with a new reader, so it can be read again by the caller.
+		resp.Body = io.NopCloser(bytes.NewBuffer(body))
```

Comment:
@DrJosh9000 Is there a better way to do this?

Reply:
So you can either do it this way (note that in this particular instance you should use `bytes.NewReader` (which is read-only) instead of `bytes.NewBuffer`) or you can use [`httputil.DumpResponse(resp, true)`](https://pkg.go.dev/net/http/httputil#DumpResponse) which does the "body swap" for you.